### PR TITLE
BUG/PERF: Sparse get_dummies uses concat

### PR DESCRIPTION
**Pull request by TomAugspurger**
Working around the DataFrame constructor perf issue in https://github.com/pandas-dev/pandas/issues/24368

Fixes deprecation warnings in the ASV files so there's something to run.

Closes https://github.com/pandas-dev/pandas/issues/24371

### CODE REVIEW

File: doc/source/whatsnew/v0.24.0.rst [hunk 1]
```diff
@@ -1613,6 +1613,7 @@ Sparse
 - Bug in :meth:`SparseArary.unique` not returning the unique values (:issue:`19595`)
 - Bug in :meth:`SparseArray.nonzero` and :meth:`SparseDataFrame.dropna` returning shifted/incorrect results (:issue:`21172`)
 - Bug in :meth:`DataFrame.apply` where dtypes would lose sparseness (:issue:`23744`)
+- Bug in :func:`concat` when concatenating a list of :class:`Series` with all-sparse values changing the ``fill_value`` and converting to a dense Series (:issue:`24371`)
```
Comment (TomAugspurger) [inline doc/source/whatsnew/v0.24.0.rst]: When the input to `concat` is a `List[Series[Sparse]]`, we now return a DataFrame with sparse values. Previously this was a dense DataFrame (probably a bug), so it isn't API breaking.

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/reshape/reshape.py [hunk 2]
```diff
@@ -909,7 +910,15 @@ def _make_col_name(prefix, prefix_sep, level):
         index = None
 
     if sparse:
-        sparse_series = {}
+
+        if is_integer_dtype(dtype):
```
[the remark this answers is not in the document]
Comment (jreback) [reply pandas/core/reshape/reshape.py]: we have that too let’s try to not reinvent the wheel

**Comment by TomAugspurger** [comment]
Do you know the method name? I don't see one that does it.

File: pandas/core/reshape/reshape.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (TomAugspurger) [reply pandas/core/reshape/reshape.py]: I didn't a function like this in any of the dtypes modules.

**Comment by TomAugspurger** [comment]
I think this should be merged soon if possible. The CI failures are blocking https://github.com/pandas-dev/pandas/pull/20796 and the explode PR.

I haven't made too much progress on fixing https://github.com/pandas-dev/pandas/issues/24368 properly. Too many edge cases in our constructors.

**Review by jreback — APPROVED** [review]

**Pull request merged**